apis/HdmiCecSource/IHdmiCecSource.h

```diff
@@ -0,0 +1,172 @@
+
+
+            struct EXTERNAL INotification : virtual public Core::IUnknown
+            {
+                enum { ID = ID_HDMI_CEC_SOURCE_NOTIFICATION };
+    
+                // @brief Device added event
+                // @text OnDeviceAdded
+                // @param logicalAddress: Logical address of the added device
+                virtual void OnDeviceAdded(const uint8_t logicalAddress) {};
+    
+                // @brief Device removed event
+                // @text OnDeviceRemoved
+                // @param logicalAddress: Logical address of the removed device
+                virtual void OnDeviceRemoved(const uint8_t logicalAddress /* @in */) {};
+    
+                // @brief Device info updated event
+                // @text OnDeviceInfoUpdated
+                // @param deviceInfo: Device info of the updated device
+                virtual void OnDeviceInfoUpdated(const uint8_t logicalAddress/* @in */) {};
+    
+                // @brief Active source status updated event
+                // @text OnActiveSourceStatusUpdated
+                // @param isActiveSource: Is the active source active or not
+                virtual void OnActiveSourceStatusUpdated(const bool status /* @in */) {};
+    
+                // @brief Triggered when a device enters standby
+                // @text StandbyMessageReceived
+                // @param logicalAddress: Logical address of the device
+                virtual void StandbyMessageReceived(const int8_t logicalAddress /* @in */) {};
+
+                // @brief Triggered when a key release message is received
+                // @text SendKeyReleaseMsgEvent
+                // @param logicalAddress: Logical address of the device
+                virtual void SendKeyReleaseMsgEvent(const int8_t logicalAddress /* @in */) {};
+
+
+                // @brief Triggered when a key press message is received
+                // @text SendKeyPressMsgEvent
+                // @param logicalAddress: Logical address of the device
+                // @param keyCode: Key code of the key press event
+                virtual void SendKeyPressMsgEvent(const int8_t logicalAddress /* @in */, const int32_t keyCode /* @in */) {};
+            };
+
+            // @json:omit
+            virtual uint32_t Register(Exchange::IHdmiCecSource::INotification *notification) = 0;
+            // @json:omit
+            virtual uint32_t Unregister(Exchange::IHdmiCecSource::INotification *notification) = 0;
+            // @json:omit
+            virtual uint32_t Configure(PluginHost::IShell* service) = 0;
+
+            /** Gets the status of the active source. */
+            // @text getActiveSourceStatus
+            // @brief Gets the status of the active source
+            // @param status: Is the active source active or not
+            // @param success: Is the operation successful or not
+            virtual uint32_t GetActiveSourceStatus(bool &status /* @out */, bool &success /* @out */) const = 0;
```

Comment:
All non-event methods should return "Core::hresult".
Ex: https://github.com/rdkcentral/entservices-apis/blob/develop/apis/RDKWindowManager/IRDKWindowManager.h#L42

Reply:
Updated to use correct return type